packages/framework-provider-aws-infrastructure/src/infrastructure/stacks/auth-stack.ts

```diff
@@ -20,12 +15,14 @@ export class AuthStack {
     private readonly apis: APIs
   ) {}
 
-  public build(): void {
+  public build(): UserPool | undefined {
     if (this.config.thereAreRoles) {
       const userPool = this.buildUserPool()
       this.buildUserPoolClient(userPool)
       this.buildAuthAPI(userPool)
+      return userPool
     }
+    return undefined
```

Comment:
We can avoid returning `undefined` explicitly, as it's what it returns implicitly if I recall correctly
```suggestion
```

Reply:
Since we have an if statement that returns is required, all routes should return a value.

Reply:
Ah I see :)